device/include/ReadAnyGroup.h

```diff
@@ -350,6 +350,9 @@ namespace ChimeraTK {
     try {
       _owner->push_elements[index].getHighLevelImplElement()->_readQueue.pop_wait();
     }
+    catch(ChimeraTK::runtime_error&) {
+      _owner->push_elements[index].getHighLevelImplElement()->_activeException = std::current_exception();
+    }
     catch(detail::DiscardValueException&) {
```

Comment:
Also boost::thread_interrupted has to be delayed here

Reply:
added, thanks for the hint